### DISCUSSION TITLE
Souce downloads from github: tar gz file sha256 file hashes have changed, fails builds

### DISCUSSION THREAD

**Discussion by ShoryuKyzan** [community/community#45855] [Actions] [answered]
### Select Topic Area

Bug

### Body

# Summary
Downloading a file from github results in a different hash than was previously recorded in the past run of a build, for the same version of the source code.
I have verified that the source code being downloaded has not changed. This is a prior release artifact (abseil-cpp 2021-11-02).
Not sure what binary change happened to the .tar.gz. This could break many build systems that verify the download.

(From https://github.com/versatica/mediasoup/issues/998)
Seeing the exact same issue as https://github.com/versatica/mediasoup/issues/762 , different hashes this time though.

# Details

Manual check of the hash:
```
wget -qO- https://github.com/abseil/abseil-cpp/archive/20211102.0.tar.gz | sha256sum
ee834b6d3de30437ac627f547dcfe791f63e61694ff68b34f75bd7b11f668343 *-
```

Contains exact same source as
https://github.com/abseil/abseil-cpp/releases/tag/20211102.0
https://github.com/abseil/abseil-cpp/tree/215105818dfde3174fe799600bb0f3cae233d0bf

Results in build error for mediasoup
```
  meson.build:199:0: ERROR: Incorrect hash for source:
   dcf71b9cba8dc0ca9940c4b316a0c796be8fab42b070bb6b7cab62b48f0e66c4 expected
   ee834b6d3de30437ac627f547dcfe791f63e61694ff68b34f75bd7b11f668343 actual.
```
Different hashes than what https://github.com/versatica/mediasoup/blob/0b770e7e226b77f8b5eb184b50583a8b73507a8b/worker/subprojects/abseil-cpp.wrap expects (the build system expects the downloaded hash to match).

We are running this build in github actions using `node:16-bullseye` image

Last time it was github messing with the .tar.gz file contents. Seems like that's happening again?

**Comment by jge162** [comment, marked as answer]
@ShoryuKyzan see here. [Closed Ticket for tar gz](https://github.com/orgs/community/discussions/45830)

**Reply by ShoryuKyzan** [reply]
thanks! i'll follow this thread then! missed it in the results somehow
